### PR TITLE
Refactor LoadEventNexus and related classes

### LINKED ISSUE #17304: Refactor LoadEventNexus and its auxiliary classes

### PR DESCRIPTION
**Pull request by wdzhou**
Description of work.

Refactor LoadEventNexus and related classes to improve readability.

**To test:**

Load

```
PG3_29571 - 974MB
NOM_77533 - 5.9GB for a couple of times.
```

Fixes #17304.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/ProcessBankData.cpp [hunk 1]
```diff
@@ -27,6 +27,36 @@ ProcessBankData::ProcessBankData(
   m_cost = static_cast<double>(numEvents);
 }
 
+void ProcessBankData::preCount() {
+
```
Comment (KarlPalmen) [inline Framework/DataHandling/src/ProcessBankData.cpp]: Wouldn't preAllocate be a better name?

**Comment by KarlPalmen** [comment]
To test. I'll need to know how to obtain the files listed.

**Comment by KarlPalmen** [comment]
I've unassigned myself to give someone else a chance to test this while I'm away for the rest of this month.

**Comment by wdzhou** [comment]
This PR has not been finished yet.  There will be more work on it until it is ready to use HDF5 library.

**Comment by wdzhou** [comment]
This approach causes a server slowdown comparing original code.  Failed.

**Comment by wdzhou** [comment]
It is closed without merging.

**Pull request closed without merging**